src/Voron/Data/VoronStream.cs

```diff
@@ -178,6 +181,19 @@ public override int Read(byte[] buffer, int offset, int count)
             return count;
         }
 
+        private Page GetPage(long pageNumber)
+        {
+            if (_encrypted == false)
+                return _llt.GetPage(pageNumber);
+
+            if (_lastPage.IsValid)
+            {
+                _llt.ForgetAboutPage(_lastPage.PageNumber);
```

Comment:
This assumes that `_lastPage` will be overwritten by the value returned by this method. What if I used this method to get  another page (and not necessary assign it to `_lastPage`)?

Reply:
What is worse, what if there is *another* piece of code that looks at this page? 
You are going to make the transaction forget about it _globally_ for everything in the transaction

Reply:
@arekpalinski changed
@ayende this is handled here:
https://github.com/ravendb/ravendb/blob/4ad2bc99fa2881b8eaa184f2a7ff0ab9d173d877/src/Voron/Impl/Paging/CryptoPager.cs#L527
We check if someone is looking at this page.